### PR TITLE
Use Github API instead of Bintray API

### LINKED ISSUE #202: Update Micronaut HTTP Client Guide to replace Bintray's api usage
we could use Github's Api. See Login with Github guide.

### PR DESCRIPTION
**Pull request by ilopmar**
Fixes #202

### CODE REVIEW

File: guides/micronaut-http-client/groovy/src/main/groovy/example/micronaut/GithubController.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        this.githubApiClient = githubApiClient
+    }
+
+    @Get("/releases-lowlevel") // <3>
+    Maybe<List<GithubRelease>> releasesWithLowLevelClient() { // <4>
+        return githubLowLevelClient.fetchReleases()
+    }
+
+    @Get(uri = "/releases", produces = MediaType.APPLICATION_JSON_STREAM) // <5>
+    Flowable<GithubRelease> packages() { // <6>
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/groovy/src/main/groovy/example/micronaut/GithubController.groovy]: method name should be releases

File: guides/micronaut-http-client/groovy/src/main/groovy/example/micronaut/GithubApiClient.groovy [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package example.micronaut
+
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.Header
+import io.micronaut.http.client.annotation.Client
+import io.reactivex.Flowable
+
+@Client(GithubConfiguration.GITHUB_API_URL) // <1>
+@Header(name = "User-Agent", value = "Micronaut HTTP Client") // <2>
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/groovy/src/main/groovy/example/micronaut/GithubApiClient.groovy]: I'm not sure how important it is, but they recommend adding `Accept: application/vnd.github.v3+json`

File: guides/micronaut-http-client/java/src/main/java/example/micronaut/GithubApiClient.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package example.micronaut;
+
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Header;
+import io.micronaut.http.client.annotation.Client;
+import io.reactivex.Flowable;
+
+@Client(GithubConfiguration.GITHUB_API_URL) // <1>
+@Header(name = "User-Agent", value = "Micronaut HTTP Client") // <2>
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/java/src/main/java/example/micronaut/GithubApiClient.java]: should use `io.micronaut.http.HttpHeaders.USER_AGENT`

File: guides/micronaut-http-client/micronaut-http-client.adoc [hunk 4]
```diff
@@ -8,7 +8,7 @@ Micronaut Version: @micronaut@
 
 == Getting Started
 
-In this guide we are going to create a Micronaut app written in @language@ to consume the Bintray API with the Micronaut HTTP Client.
+In this guide we are going to create a Micronaut app written in @language@ to consume the Github API with the Micronaut HTTP Client.
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/micronaut-http-client.adoc]: Should be GitHub

File: guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Test
+    fun verifyGithubReleasesCanBeFetchedWithLowLevelHttpClient() {
+        //when:
+        val request: HttpRequest<Any> = HttpRequest.GET("/github/releases-lowlevel")
+        val rsp = client.toBlocking().exchange(request, // <3>
+            Argument.listOf(GithubRelease::class.java)) // <4>
+
+        //then: 'the endpoint can be accessed'
+        Assertions.assertEquals(HttpStatus.OK, rsp.status) // <5>
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt]: should use static imports

File: guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        //then:
+        for (name in expectedReleases) {
+            Assertions.assertTrue(StreamSupport.stream(githubReleases.spliterator(), false)
+                .map(GithubRelease::name)
+                .anyMatch { anObject: String? -> name.equals(anObject) })
+        }
+    }
+
+    companion object {
+        private val expectedReleases = Arrays.asList("Micronaut 2.5.0", "Micronaut 2.4.4", "Micronaut 2.4.3")
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt]: more idiomatic to use `listOf()`

File: guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+import io.micronaut.core.type.Argument
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.client.RxStreamingHttpClient
+import io.micronaut.http.client.annotation.Client
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import java.util.*
```
Comment (burtbeckwith) [inline guides/micronaut-http-client/kotlin/src/test/kotlin/example/micronaut/GithubControllerTest.kt]: should be just `java.util.Arrays` (or remove if using `listOf()` below)

**Comment by ilopmar** [comment]
Thanks for the good feedback @burtbeckwith. I've done all the changes. The build failed yesterday when I submitted the PR but the tests pass locally. Let's see if this new run works...

**Comment by ilopmar** [comment]
Tests fixed. The problem was that we have in `application.yml`:
```
github:
  repository: micronaut-core
```

And when running on CI, GitHub Actions automatically inject the environment variable `GITHUB_REPOSITORY` with the current repository. That of course overrides the config in application.yml and the tests send the request to the wrong repository.
```
Sending HTTP GET to https://api.github.com/repos/micronaut-projects/micronaut-projects/micronaut-guides-poc/releases
```

I've rename `repository` to `repo` in the config and in the POJOs.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**